server/pbench/bin/pbench-server-activate-create-crontab

```diff
@@ -21,31 +21,43 @@ mkdir -p $crontabpath
 chmod 755 $crontabpath
 crontab=$crontabpath/crontab
 
-if [ ! -f $CONFIG ]
+> $crontab
+
+if [ ! -f "$CONFIG" ]
 then
     echo "Config file $CONFIG does not exist"
     exit 2
 fi
 
-echo "CONFIG=$CONFIG" > $crontab
 bindir=$(getconf.py script-dir pbench-server)
 if [ -z "$bindir" ] ;then
     echo "No 'script-dir' setting in [pbench-server] section of config file $CONFIG"
     exit 3
 fi
-echo "PATH=$bindir:$PATH" >> $crontab
 
+# what roles does this server play?
 roles=$(getconf.py -l roles pbench-server)
+
+# the hostname of the host we are installing in.
 hostname=$(hostname -f)
-# in some cases, the host specified in the config file
+
+# In some cases, the host specified in the config file
 # is different from what hostname -f returns. In such
 # cases, we also have a "real" name in the config file
-# that *should* match what hostname -f returns
+# that *should* match what hostname -f returns, otherwise
+# the sanity check below will just skip all the roles for
+# the host, leaving us with an empty crontab. If you do
+# get an empty crontab, this is a good thing to check.
 realhost=$(getconf.py realhost pbench-server)
 
-function crontab_normal() {
-    local role=$1
-    local host mailfrom mailto
+function crontab_header() {
+    role=$1
+    local mailfrom mailto
+
+    grep -q PATH= $crontab ||
+        echo "PATH=$bindir:$PATH" >> $crontab
+    grep -q CONFIG $crontab ||
+        echo "CONFIG=$CONFIG" >> $crontab
```

Comment:
For these two statements, can you add a comment explaining what they do?  It seems subtle enough to warrant one.